contracts/interchain-token-service/src/contract.rs

```diff
@@ -539,8 +523,59 @@ impl InterchainTokenService {
                     );
                 }
             }
-            Message::DeployInterchainToken(_) => {
-                todo!()
+            Message::DeployInterchainToken(DeployInterchainToken {
+                token_id,
+                name,
+                symbol,
+                decimals,
+                minter,
+            }) => {
+                ensure!(!name.is_empty(), ContractError::EmptyTokenName);
```

Comment:
We can use `validate_token_metadata`. to validate name, symbol, decimals. reference: https://github.com/axelarnetwork/axelar-cgp-soroban/blob/6ec26221bd6a7583b65bde93a2f69a7abb4dacb9/contracts/interchain-token-service/src/contract.rs#L295-L301